.NET/Microsoft.Recognizers.Definitions.Common/English/NumbersDefinitions.cs

```diff
@@ -54,8 +54,10 @@ public static class NumbersDefinitions
       public const string RelativeOrdinalRegex = @"(?<relativeOrdinal>(next|previous|current)\s+one|(the\s+second|next)\s+to\s+last|the\s+one\s+before\s+the\s+last(\s+one)?|the\s+last\s+but\s+one|(ante)?penultimate|last|next|previous|current)";
       public static readonly string BasicOrdinalRegex = $@"({NumberOrdinalRegex}|{RelativeOrdinalRegex})";
       public static readonly string SuffixBasicOrdinalRegex = $@"(?:(((({TensNumberIntegerRegex}(\s+(and\s+)?|\s*-\s*){ZeroToNineIntegerRegex})|{TensNumberIntegerRegex}|{ZeroToNineIntegerRegex}|{AnIntRegex})(\s+{RoundNumberIntegerRegex})+)\s+(and\s+)?)*({TensNumberIntegerRegex}(\s+|\s*-\s*))?{BasicOrdinalRegex})";
+      public static readonly string SuffixNumberOrdinalRegex = $@"(?:(((({TensNumberIntegerRegex}(\s+(and\s+)?|\s*-\s*){ZeroToNineIntegerRegex})|{TensNumberIntegerRegex}|{ZeroToNineIntegerRegex}|{AnIntRegex})(\s+{RoundNumberIntegerRegex})+)\s+(and\s+)?)*({TensNumberIntegerRegex}(\s+|\s*-\s*))?{NumberOrdinalRegex})";

```

Comment:
Why did you have to introduce a new regex?

Reply:
The original setup was as follows: 

`FractionNounWithArticleRegex` (the regex causing this bug) included `AllOrdinalRegex` which makes use of `SuffixBasicOrdinalRegex` to include regexes for all ordinals. This includes the relative ordinals (next, previous etc.) that we do not want to include.

I wanted to create a parallel setup without the inclusion of the relative ordinals. Also I did not simply replace `AllOrdinalRegex` because it is used in other areas. Since it uses `SuffixBasicOrdinalRegex`, I could not replace that either, so I just created a new pair of regexes for only number and rounded number ordinals. They are:

`SuffixNumberOrdinalRegex` and `AllNumberOrdinalRegex` (since this only includes number ordinals and rounded number ordinals without including relative ordinals).

Please let me know if there is an alternative approach or better solution that I can adopt instead.

Reply:
Perhaps this can be extended to handle #2826, if it doesn't already.
If it works, let's add the relevant specs.

Reply:
@aitelint Is #2826 on your list?

Reply:
yes, but it seems it is possibly addressed by this PR. Should I look into it?

Reply:
Please do and validate if the current approach has perf impact that could be solved differently. Thanks.